### PR TITLE
fix an issue where raw json output is written to IDE clients

### PR DESCRIPTION
**Pull request by devoncarew**
## Description

This addresses an issue where an error message and raw json output is sent to IDE clients (the output is generally meant to be part of the flutter tool / IDE daemon protocol); https://github.com/flutter/flutter/issues/61361:

Screen Shot 2020-09-08 at 8 20 26 PM

This was likely due to a refactoring of the logging logic. It's possible to send progress messages (for things like installing portions of the flutter sdk, or running flutter pub get) before the app has started running; after the refactoring these progress messages are sent as errors if they happen before the app has started.

I don't have a test for this, but would love guidance on where a test should live, and which existing tests would be good to model off of.

## Related Issues

- https://github.com/flutter/flutter/issues/61361

## Tests

I added the following tests:

- in `test/general.shard/base/logger_test.dart`, added `'AppRunLogger writes plain text statuses when no app is active'`

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

@jonahwilliams @zanderso

### CODE REVIEW

**Comment by devoncarew** [comment]
Test added (thanks to @DanTup, https://github.com/DanTup/flutter/commit/829f278640e5d671d2ff784c280cf7f1ec38c611). PTAL -

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 1]
```diff
@@ -1105,8 +1111,8 @@ class AppInstance {
   }
 
   Future<T> _runInZone<T>(AppDomain domain, FutureOr<T> method()) async {
-    _logger.domain = domain;
-    _logger.app = this;
+    // _logger.domain = domain;
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/daemon.dart:1114]: nit: remove commented code

**Review by jonahwilliams — APPROVED** [review]
LGTM, thanks @devoncarew !

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 1 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/commands/daemon.dart]: Done!

**Pull request merged**